### PR TITLE
(2288) Show multiple regulators and roles on professions page

### PR DESCRIPTION
**Pull request by pezholio**
This alters the regulator show pages to show their role above their details, and their name (as well as a link to the regulator page) in the summary list below

### Before

image

### After

image

### CODE REVIEW

File: src/professions/admin/profession-versions.controller.spec.ts [hunk 1]
```diff
@@ -154,7 +157,16 @@ describe('ProfessionVersionsController', () => {
     describe('when the Profession is complete', () => {
       describe('when the Profession has a single Organisation', () => {
         it('should return populated template params', async () => {
-          const profession = professionFactory.build();
+          const organisation = organisationFactory.build();
+
+          const profession = professionFactory.build({
+            professionToOrganisations: [
```
Comment (Gweaton) [inline src/professions/admin/profession-versions.controller.spec.ts]: I've been wondering if this is something we should create a new factory for to save us having to do `as ProfessionToOrganisation` but I don't know if it saves us all that many lines of code - it might just be a bit easier to read?

File: src/professions/professions.controller.ts [hunk 2]
```diff
@@ -41,17 +40,9 @@ export class ProfessionsController {
...
-        const organisation = Organisation.withLatestLiveVersion(
-          professionToOrganisation.organisation,
-        );
-
-        return {
-          ...organisation,
-          role: professionToOrganisation.role,
-        };
-      },
+    const tierOneOrganisations = getGroupedTierOneOrganisationsFromProfession(
```
Comment (Gweaton) [inline src/professions/professions.controller.ts:43]: Really nice to have that old logic extracted into this helper.

File: views/professions/show.njk.spec.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { JSDOM } from 'jsdom';
```
Comment (Gweaton) [inline views/professions/show.njk.spec.ts:1]: Interesting - I've never written view specs in JS before. When do you think we should opt for these over e2e tests? Is it just when we have too much data for seeds, or is it to reduce the overall number of e2e tests which are fairly slow and expensive to maintain?

Just wondering about how we make this decision going forward

**Review by Gweaton — APPROVED** [review]
Well done - this looks like it was a challenging task! I think this looks really good - just have a few questions.

File: views/professions/show.njk.spec.ts (hunk 3 above)
Comment (pezholio) [reply views/professions/show.njk.spec.ts:1]: Yeah, I think it's just to reduce the overload of having seeds for every single edge case. Also changing seeds often breaks other tests which we then have to fix. I do wonder if it might be prudent to have specific seeds for specific test cases, but I dunno.

File: cypress/integration/organisations/search/search.spec.ts [hunk 4]
```diff
@@ -252,10 +252,11 @@ function findMatchingOrganisations(filter: (profession: any) => boolean) {
...
     const organisationNames = matchingProfessions
       .map((profession) => [
-        profession.organisation,
-        profession.additionalOrganisation,
+        profession.organisations.map((organisation) => organisation.name),
       ])
-      .flat();
+      .flat(Infinity);
+
+    console.log(organisationNames);
```
Comment (richpjames) [inline cypress/integration/organisations/search/search.spec.ts:260]: should this log be here?

Comment (richpjames) [inline cypress/integration/organisations/search/search.spec.ts:260]: should this log be here?

Comment (pezholio) [reply cypress/integration/organisations/search/search.spec.ts:260]: Oooh, good catch! A relic of debugging somewhere

File: src/professions/admin/profession-versions.controller.spec.ts (hunk 1 above)
Comment (pezholio) [reply src/professions/admin/profession-versions.controller.spec.ts]: Yeah, that's exactly the conversation I had in my head, I dunno if it's worth it. We do have that `organisations` transient param, but that doesn't allow us to specify the roles, and I dunno if it's worth extending it either.
